aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -497,6 +497,8 @@ module aptos_framework::coin {
     }
 
     public fun register<CoinType>(account: &signer) {
+        assert!(is_coin_initialized<CoinType>(), error::invalid_argument(ECOIN_INFO_NOT_PUBLISHED));
```

Comment:
Doesn't this serialize registers on looking up if the coin info exists?

Reply:
just to be clear, we're going to add in a new read resource for each time we create an account or initialize a coin

I think that's probably why it was left out

Reply:
This shouldn't a problem now with reduced gas

Reply:
Based on the land-blocking forge test, there doesn't seem to be any execution degradation that affects P2P tx throughput (relatively to the test on another recent PR)